data-prepper-core/src/main/java/com/amazon/dataprepper/pipeline/server/DataPrepperServer.java

```diff
@@ -0,0 +1,36 @@
+package com.amazon.dataprepper.pipeline.server;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import com.sun.net.httpserver.HttpServer;
+
+/**
+ * Class to handle any serving that the data prepper instance needs to do.
+ * Currently, only serves metrics in prometheus format.
+ */
+public class DataPrepperServer {
+
+    private final HttpServer server;
+
+    public DataPrepperServer(final int port) throws IOException {
+        server = HttpServer.create(
```

Comment:
This works for now but we might have to consider having a separate named/fixed threadpool for this server. We can work on it as we build more things around this.

Reply:
Yep, for sure.